trie/trie_test.go

```diff
@@ -74,175 +87,218 @@ func TestNewTrie(t *testing.T) {
+		{key: []byte{0x07}, value: []byte("ramen"), op: PUT},
+		{key: []byte{0xf2}, value: []byte("pho"), op: PUT},
+		{key: []byte("noot"), value: nil, op: GET},
+		{key: []byte{0}, value: nil, op: GET},
+		{key: []byte{0x01, 0x35}, value: []byte("spaghetti"), op: GET},
+		{key: []byte{0x01, 0x35, 0x79}, value: []byte("gnocchi"), op: GET},
+		{key: []byte{0x07}, value: []byte("ramen"), op: GET},
+		{key: []byte{0xf2}, value: []byte("pho"), op: GET},
 	}
 
-	err = trie.Put(key2, value2)
-	if err != nil {
-		t.Errorf("Fail to put with key %x and value %x: %s", key2, value2, err.Error())
-	}
+	runTests(t, trie, tests)
+}
 
-	err = trie.Put(key3, value3)
-	if err != nil {
-		t.Errorf("Fail to put with key %x and value %x: %s", key3, value3, err.Error())
-	}
+func TestPutAndGetOddKeyLengths(t *testing.T) {
+	trie := newEmpty()
 
-	err = trie.Put(key4, value4)
-	if err != nil {
-		t.Errorf("Fail to put with key %x and value %x: %s", key4, value4, err.Error())
+	tests := []trieTest{
+		{key: []byte{0x43, 0xc1}, value: []byte("noot"), op: PUT},
+		{key: []byte{0x49, 0x29}, value: []byte("nootagain"), op: PUT},
+		{key: []byte{0x43, 0x0c}, value: []byte("odd"), op: PUT},
+		{key: []byte{0x4f, 0x4d}, value: []byte("stuff"), op: PUT},
+		{key: []byte{0x4f, 0xbc}, value: []byte("stuffagain"), op: PUT},
+		{key: []byte{0x43, 0xc1}, value: []byte("noot"), op: GET},
+		{key: []byte{0x49, 0x29}, value: []byte("nootagain"), op: GET},
+		{key: []byte{0x43, 0x0c}, value: []byte("odd"), op: GET},
+		{key: []byte{0x4f, 0x4d}, value: []byte("stuff"), op: GET},
+		{key: []byte{0x4f, 0xbc}, value: []byte("stuffagain"), op: GET},
 	}
 
-	val, err := trie.Get([]byte{0})
-	if err != nil {
-		t.Errorf("Fail to get key %x: %s", key1, err.Error())
-	} else if !bytes.Equal(val, nil) {
-		t.Errorf("Fail to get key %x with nil value: got %x", "noot", val)
-	}
+	runTests(t, trie, tests)
+}
 
-	val, err = trie.Get(key1)
-	if err != nil {
-		t.Errorf("Fail to get key %x: %s", key1, err.Error())
-	} else if !bytes.Equal(val, value1) {
-		t.Errorf("Fail to get key %x with value %x: got %x", key1, value1, val)
+func TestPutAndGet(t *testing.T) {
+	for i := 0; i < 10; i++ {
+		trie := newEmpty()
+		rt := generateRandTest(1000)
+		for _, test := range rt {
+			err := trie.Put(test.key, test.value)
+			if err != nil {
+				t.Errorf("Fail to put with key %x and value %x: %s", test.key, test.value, err.Error())
+			}
+
+			val, err := trie.Get(test.key)
+			if err != nil {
+				t.Errorf("Fail to get key %x: %s", test.key, err.Error())
+			} else if !bytes.Equal(val, test.value) {
+				t.Errorf("Fail to get key %x with value %x: got %x", test.key, test.value, val)
+			}
+		}
+
+		for _, test := range rt {
+			val, err := trie.Get(test.key)
+			if err != nil {
+				t.Errorf("Fail to get key %x: %s", test.key, err.Error())
+			} else if !bytes.Equal(val, test.value) {
+				t.Errorf("Fail to get key %x with value %x: got %x", test.key, test.value, val)
+				writeToTestFile(rt)
```

Comment:
Only the last failing case will be written to file

Reply:
no, the entire test is written to the file. rt is the array of all the kv pairs

Reply:
nvm i get it, fixed